README.md
# Alicia

It's a file uploader and manager with below features:

- Upload pictures and videos and files
- Store external links
- HLS support
- Collect file's detail
- Classification for files
- Automatic optimization

for more information [see documentation](https://docs-alicia.vercel.app/).

## Installation

Install the package via composer

```shell
composer require hans-thomas/alicia
```

Then, publish config file using

```shell
php artisan vendor:publish --tag alicia-config
```

In the end, use `AliciaHandler` trait on your model.

```php
use Hans\Alicia\Traits\AliciaHandler;
use Illuminate\Database\Eloquent\Model;

class Post extends Model {
    use AliciaHandler;
    
    // ...
}
```

Support
-------

- [Documentation](https://docs-alicia.vercel.app/)
- [Report bugs](https://github.com/hans-thomas/alicia/issues)



docs/CHANGELOG.md
### Changelog

All notable changes to this project will be documented in this file. Dates are displayed in UTC.

Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).

#### [v0.5.0](https://github.com/h-enk/doks/compare/v0.4.3...v0.5.0)

- Better multilingual support [`#818`](https://github.com/h-enk/doks/pull/818)
- 🧪 Add scrollspy to toc pages as an option (experimental) [`#797`](https://github.com/h-enk/doks/pull/797)
- Better blog pages [`#798`](https://github.com/h-enk/doks/pull/798)
- Better image support [`#809`](https://github.com/h-enk/doks/pull/809)
- feat(search): allow complete section customization [`#799`](https://github.com/h-enk/doks/pull/799)
- fix(menus): sort pages and sections by default order [`#796`](https://github.com/h-enk/doks/pull/796)
- Add to top button as an option [`#794`](https://github.com/h-enk/doks/pull/794)
- Single navbar [`#782`](https://github.com/h-enk/doks/pull/782)
- Use SVG favicon if it exists [`#754`](https://github.com/h-enk/doks/pull/754)
- Add custom-head.html [`#755`](https://github.com/h-enk/doks/pull/755)
- fix(layouts): index.redirects truncates newline [`#775`](https://github.com/h-enk/doks/pull/775)
- fix: csp invalid for global alert dismissal [`#788`](https://github.com/h-enk/doks/pull/788)
- deps: bump versions to
  latest [`79a0449`](https://github.com/h-enk/doks/commit/79a0449f15f0b9a3b77ee6d98a8d06e6fd94970b)
- deps: bump versions to
  latest [`b1dfb71`](https://github.com/h-enk/doks/commit/b1dfb71f9020d471611c32d847a2c54410819718)
- feat: add all in single
  navbar [`16152d6`](https://github.com/h-enk/doks/commit/16152d63609c5722186087742ad821779fb20fc0)

#### [v0.4.3](https://github.com/h-enk/doks/compare/v0.4.2...v0.4.3)

> 18 May 2022

- fix(flexsearch): separate chain of calls into statements [`#739`](https://github.com/h-enk/doks/pull/739)
- deps: bump versions to
  latest [`f8bcce9`](https://github.com/h-enk/doks/commit/f8bcce9e3ca966ff894e5c2f9be6324db99e23ce)
- meta: update changelog [`cf51ea8`](https://github.com/h-enk/doks/commit/cf51ea887dd12f79588f950988d0eb6c6895407f)

#### [v0.4.2](https://github.com/h-enk/doks/compare/v0.4.1...v0.4.2)

> 1 April 2022

- ux: fix details for firefox [`d4b6900`](https://github.com/h-enk/doks/commit/d4b6900c965da251d02834bf4a7eb17e6f12e3d2)
- meta: update content security
  policy [`7c834a5`](https://github.com/h-enk/doks/commit/7c834a5a11858f8eb47fca558d3eb620589f1275)
- ux: slide offcanvas menu from right to
  left [`f13cb0e`](https://github.com/h-enk/doks/commit/f13cb0e8de97d37c4a23e0062aeedc8cb84cb0ab)

#### [v0.4.1](https://github.com/h-enk/doks/compare/v0.4.0...v0.4.1)

> 30 March 2022

- fix(layouts): use the permalink instead of url [`#710`](https://github.com/h-enk/doks/pull/710)
- fix(layouts): use comma for more than 2 contributors [`#714`](https://github.com/h-enk/doks/pull/714)
- Adds details shortcode [`#661`](https://github.com/h-enk/doks/pull/661)
- deps: uninstall mermaid [`a927710`](https://github.com/h-enk/doks/commit/a92771068520221d33de724bd1f971760c640e5e)
- deps: bump versions to
  latest [`d127ee6`](https://github.com/h-enk/doks/commit/d127ee60b29915aff22f3ee4504f63db5c297847)
- feat: add details shortcode [`7bc15dc`](https://github.com/h-enk/doks/commit/7bc15dc0ec65b48b98f9150f72a2abf7a681ab1f)

#### [v0.4.0](https://github.com/h-enk/doks/compare/v0.3.5...v0.4.0)

> 4 February 2022

- Adds scroll position lock for default docs sidebar [`#648`](https://github.com/h-enk/doks/pull/648)
- feat: edit link docs repo subpath [`#618`](https://github.com/h-enk/doks/pull/618)
- Sets font weight active page to 500 [`#644`](https://github.com/h-enk/doks/pull/644)
- Adds support for sub navigation main menu [`#642`](https://github.com/h-enk/doks/pull/642)
- [#604] Allow Firefox on macOS to see italic text. [`#626`](https://github.com/h-enk/doks/pull/626)
- Adds new docs tree — with a single command 🤖  [`#637`](https://github.com/h-enk/doks/pull/637)
- Adds full width as an option [`#636`](https://github.com/h-enk/doks/pull/636)
- chore(deps-dev): bump clipboard from 2.0.8 to 2.0.9 [`#635`](https://github.com/h-enk/doks/pull/635)
- feat: add multi level section menu [`#617`](https://github.com/h-enk/doks/pull/617)
- feat: changeable global alert + fix: cumulative layout shift [`#613`](https://github.com/h-enk/doks/pull/613)
- chore(deps-dev): bump mermaid from 8.13.2 to 8.13.3 [`#513`](https://github.com/h-enk/doks/pull/513)
- Improve flexsearch feedback [`#517`](https://github.com/h-enk/doks/pull/517)
- Simplify heading rendering [`#516`](https://github.com/h-enk/doks/pull/516)
- Plainify flexsearch index values [`#511`](https://github.com/h-enk/doks/pull/511)
- Add .Summary as fallback for description [`#508`](https://github.com/h-enk/doks/pull/508)
- Fix flexsearch arrow key navigation error [`#509`](https://github.com/h-enk/doks/pull/509)
- ✨Adds additional option for 'edit this page' feature [`#496`](https://github.com/h-enk/doks/pull/496)
- Merge upstream [`#2`](https://github.com/h-enk/doks/pull/2)
- deps: bump versions to
  latest [`e2b4ab2`](https://github.com/h-enk/doks/commit/e2b4ab23df16d91d7171bc9977eee6750937d2d6)
- deps: bump versions to
  latest [`66b4b17`](https://github.com/h-enk/doks/commit/66b4b172597893735f2c8d572226bb556a1ce85f)
- deps: bump versions to
  latest [`8b17636`](https://github.com/h-enk/doks/commit/8b176366757eaed1f7ce9a9e1ad0127ff1620a48)

#### [v0.3.5](https://github.com/h-enk/doks/compare/v0.3.4...v0.3.5)

> 5 October 2021

- feat: add versioning setup [`#478`](https://github.com/h-enk/doks/pull/478)
- feat: adjust latex svg image to dark theme [`#460`](https://github.com/h-enk/doks/pull/460)
- fix: edit-page multi-language support [`#459`](https://github.com/h-enk/doks/pull/459)
- feat: add docsRepoBranch config [`#458`](https://github.com/h-enk/doks/pull/458)
- feat: image max width [`#457`](https://github.com/h-enk/doks/pull/457)
- fix: docs url [`#456`](https://github.com/h-enk/doks/pull/456)

config/config.php
<?php

	use Illuminate\Support\Carbon;
	use Illuminate\Support\Str;

	return [

		/*
		|--------------------------------------------------------------------------
		| base folder
		|--------------------------------------------------------------------------
		|
		| the base folder is where Resources upload and classification files
		|
		*/
		'base'           => base_path( 'vault' ),

		/*
		|--------------------------------------------------------------------------
		| classification
		|--------------------------------------------------------------------------
		|
		| the method of organization uploaded files
		|
		*/
		'classification' => Str::slug( Carbon::now()->toFormattedDateString() ),


		/*
		|--------------------------------------------------------------------------
		| extensions
		|--------------------------------------------------------------------------
		|
		| determine the allowed extensions
		|
		*/
		'extensions'     => [
			'images' => [ 'jpg', 'jpeg', 'png' ],
			'videos' => [ 'mp4', 'mkv' ],
			'audios' => [ 'mp3' ],
			'files'  => [ 'zip', 'rar', 'pdf', 'csv', 'xlsx', 'docx' ]
		],

		/*
		|--------------------------------------------------------------------------
		| optimization
		|--------------------------------------------------------------------------
		|
		| images and videos optimization can enable/disable independently
		|
		*/
		'optimization'   => [
			'images' => true,
			'videos' => true,
		],

		/*
		|--------------------------------------------------------------------------
		| Naming files
		|--------------------------------------------------------------------------
		|
		| determine how to generate files name.
		|
		| supported drivers: [ 'uuid', 'string', 'digits', 'string_digits', 'symbols', 'hash' ]
		|
		*/
		'naming'         => 'uuid',

		/*
		|--------------------------------------------------------------------------
		| Download link
		|--------------------------------------------------------------------------
		|
		| if you want to customize the download url, you can update this without
		| manipulating parameters
		|
		*/
		'link'           => '/download/{resource}/{hash?}', // dont add or remove parameters

		/*
        |--------------------------------------------------------------------------
        | Signing download link
        |--------------------------------------------------------------------------
        |
        | if you want a simple and permanent link just set signed key to false,
        | however if you want to protect your links leave it true
        |
        */
		'signed'         => true,

		/*
		|--------------------------------------------------------------------------
		| Secret key
		|--------------------------------------------------------------------------
		|
		| the secret key that Signature class encode and decode hashes
		|
		*/
		'secret'         => 'resource_key',

		/*
		|--------------------------------------------------------------------------
		| Link Expiration
		|--------------------------------------------------------------------------
		|
		| determine the expiration time of each link in minutes
		|
		*/
		'expiration'     => 30,

		/*
		|--------------------------------------------------------------------------
		| Route attributes
		|--------------------------------------------------------------------------
		|
		| you can define your custom attributes, like prefix etc.
		|
		*/
		'attributes'     => [
			'middleware' => [ 'api' ]
		],

		/*
		|--------------------------------------------------------------------------
		| HLS settings
		|--------------------------------------------------------------------------
		|
		| some parameter to customize HLS export
		|
		*/
		'hls'                => [
			'enable'              => true,
			'bitrate'             => [ 250, 500, 1000 ],
			'setSegmentLength'    => 10,
			'setKeyFrameInterval' => 48,
		],

		/*
		|--------------------------------------------------------------------------
		| Image exports settings
		|--------------------------------------------------------------------------
		|
		| define your resolutions in [ $height => $width ] format,
		| you can set false for disabling this feature
		|
		*/
		'export'             => [
			960  => 540,
			1280 => 720,
			1920 => 1080,
		],
	];
